SchemaValidator/SchemaValidation.py

```diff
@@ -0,0 +1,218 @@
+            # If we get this far, the XML has validated successfully.
+            return ('Valid')
+        except etree.XMLSyntaxError as err:
+            return ('Invalid:\t' + str(err))
+
+    except Exception as e:
+        print('Unexpected problem during xml validation.')
+        print(str(e))
+        print(exc_info())
+        return (str(e) + '\n' + str(exc_info()))
+
+def updateValidationStatus(uploadID=None, isValid=None, validationFeedback=None, token=None, url=None):
+    if(url is None):
+        url=getUrl()
+    try:
+        fullUrl = str(url) + '/api/uploads/' + str(uploadID) + '/'
+        # TODO: Haml Files?
+        body = {'valid': isValid, 'validationFeedback': validationFeedback, 'id':uploadID, 'type':'HML'}
+        encodedJsonData = str(json.dumps(body)).encode('utf-8')
+        updateRequest = request.Request(url=fullUrl, data=encodedJsonData, method='PUT')
+        updateRequest.add_header('Content-Type', 'application/json')
+        updateRequest.add_header('Authorization', 'Bearer ' + token)
+        responseData = request.urlopen(updateRequest).read().decode("UTF-8")
+        response=json.loads(responseData)
+
+        # The response contains the saved data, if successful. If the response matches what we expect it was a success.
+        if(str(response['id'])==str(uploadID) and str(response['valid'])==str(isValid)
+            and str(response['validationFeedback'])==str(validationFeedback)):
+            return True
+        else:
+            print('Could not set validation status, response:\n' + str(response))
+            return False
+    except SyntaxError as e:
+        print('Syntax error when parsing response from request:\n' + str(e) + '\n' + str(exc_info()))
+        return False
+    except urllib.error.HTTPError as e:
+        print('HTTP error when setting validation status for upload id ' + str(uploadID) + ' : ' + str(e))
+        return False
+    except Exception as e:
+        print('Error when updating validation status:\n' + str(e) + '\n' + str(exc_info()))
+        return False
+
+def getUploadID(uploadFileName=None, token=None, url=None):
+    print('Trying to find upload ID for file name:' + str(uploadFileName))
+    if(url is None):
+        url=getUrl()
+    if(token is None):
+        token = getToken()
+        if token is None or len(token) < 1:
+            print('Failed to get login token.')
+            return None
+    try:
+        fullUrl = str(url) + '/api/uploads/'
+        #body = {'valid': isValid, 'validationFeedback': validationFeedback, 'id':uploadID, 'type':'HML'}
+        encodedJsonData = str(json.dumps({})).encode('utf-8')
+        updateRequest = request.Request(url=fullUrl, data=encodedJsonData, method='GET')
+        updateRequest.add_header('Content-Type', 'application/json')
+        updateRequest.add_header('Authorization', 'Bearer ' + token)
+        responseData = request.urlopen(updateRequest).read().decode("UTF-8")
```

Comment:
Pretty sure, the management app would need adjustment, so the system user can also get a full list of uploads.